### ISSUE TITLE
Canvas creation is broken

### ISSUE THREAD

**Issue by Mikhail158** [Codewars/codewars.com#629]
I'm using canvases in my [level](https://www.codewars.com/kata/573b2776f5ce3fa09d000162)
But after some update they stopped working.
After some researches i found that it is because of new logging system. My 'drawer' script, that was added before test can't be used. Then i tryed to move script to create it on each level, but now it draws all canvases in single place instead of drawing them for each test individualy

**Comment by jhoffner** [comment]
Yeah its odd, for some reason even though the scripts are being executed within `eval` by the templating engine, the variables don't seem to be globally scoped. I fixed it for you for now, by assigning your common code to the window object instead. I'll rename this and leave it open so that I remember to dig deeper.

**Comment by kazk** [comment]
Isn't this because of the strict mode?
Some update last year suddenly broke many of myjinxin2015's canvas kata so I helped him fix. The issues was `ReferenceError` by forgetting some `var`.

In strict mode, `eval` doesn't make evaluated variables available in surrounding scope.

- [Strict mode (MDN)](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Strict_mode#Strict_mode_for_scripts)
- [New ES5 strict mode support: new vars created by strict mode eval code are local to that code only](http://whereswalden.com/2011/01/10/new-es5-strict-mode-support-new-vars-created-by-strict-mode-eval-code-are-local-to-that-code-only/)

```js
(function() { // normal/sloppy mode
  var x = 1;
  console.log(eval("x"));            //=> 1
  console.log(eval("var x = 4; x")); //=> 4
  console.log(x);                    //=> 4 changed
  console.log(eval("var y = 3; y")); //=> 3
  console.log(y);                    //=> 3 defined
})();

(function() {
  "use strict";
  var x = 1;
  console.log(eval("x"));            //=> 1
  console.log(eval("var x = 4; x")); //=> 4
  console.log(x);                    //=> 1 unchanged
  console.log(eval("var y = 3; y")); //=> 3
  console.log(y);                    //=> ReferenceError: y is not defined
})();
```
